pkg/util/framer/framer.go

```diff
@@ -131,13 +131,13 @@ func NewJSONFramedReader(r io.ReadCloser) io.ReadCloser {
 func (r *jsonFrameReader) Read(data []byte) (int, error) {
 	// Return whatever remaining data exists from an in progress frame
 	if n := len(r.remaining); n > 0 {
-		if n <= cap(data) {
+		if n <= len(data) {
```

Comment:
Hmm.. but if you are calling Read for the first time and your buffer initially contains no data, then its len() will be 0 (or am I missing something).
And potentially I can imagine a situation that user of this method is passing a new slice every time he calls this method (I agree that wouldn't be a smart user, but still). In that case, len(data) would always be 0, right?

I'm probably missing something, but just want to understand this change.